app/assets/javascripts/initializers/initializeSheCodedArticles.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+// This file is specifically used for SheCoded,
+// We can try to make it more general for campaigns in the future once we know more (classnames, params).
+
+function renderArticles(articles, articleElement ) {
+  let articleDiv = articleElement;
+  const articleHTML = articles.map((article) => {
+    return `<a class="she-coded-page__some-story-container-link" href="${article.url}"><div class="she-coded-page__some-story-container">
+      <div class="she-coded-page__some-story-tag"><span class="she-coded-page__some-story-hashtag">#</span>shecoded</div>
+      <div class="she-coded-page__some-story-header">${article.title}</div>
+      <div class="she-coded-page__some-story-info">
+        <div class="she-coded-page__some-story-info-author-details">
+          <div class="she-coded-page__some-story-info-pic">
+            <img src="${article.user.profile_image}" alt="${article.user.name}'s profile picture"/>
+          </div>
+          <div class="she-coded-page__some-story-info-name">${article.user.name}</div>
+        </div>
+        <div class="she-coded-page__some-story-info-date">&nbsp; &#183; &nbsp; ${article.readable_publish_date}</div>
+      </div>
+    </div></a>`
+  });
+
+  articleDiv.innerHTML = articleHTML.join(" ");
+}
+
+function renderNoArticles(articleElement) {
+  let articleDiv = articleElement;
+  articleDiv.innerHTML = "<div>There are no articles to show.</div>";
+}
+
+function getArticles(articleElement) {
```

Comment:
I think [fetch()](https://javascript.info/fetch) would result in a lot less code here. We already use it around the JS code if you need examples.

Reply:
+1, we should definitely be using fetch here